### ISSUE TITLE
Enfore the use of the @Override annotation

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/phoenicis#1663]
Currently a lot of classes override or implement methods of their parent class or interfaces without marking said methods with an `@Override` annotation.
This makes it hard for developers to know where the method has been defined.

To make our code cleaner and to prevent such situations from occurring I suggest to enforce the use of the `@Override` annotation.
Ideally we should do this with either a maven plugin or a linter.

**Comment by plata** [comment]
Do you know a plugin/linter which is capable of this?

**Comment by madoar** [comment]
No I only know that the IDE itself is able to enforce this.

**Comment by plata** [comment]
What about https://code.revelc.net/warbucks-maven-plugin/?

**Comment by qparis** [comment]
I think we should not have a maven plugin for each coding rule we decide. This is definetely a Codacy responsibility

**Comment by plata** [comment]
Seems that checkstyle can only validate this if `@inheritDoc` is used: http://checkstyle.sourceforge.net/config_annotation.html#MissingOverride

**Comment by madoar** [comment]
I agree with @qparis, I also believe that this should be something Codacy should check for us.
